src/olympia/amo/tests/test_celery.py

```diff
@@ -1,67 +1,129 @@
+import time
 import datetime
 
 from datetime import timedelta
+from unittest import mock
 
+from django.conf import settings
 from django.core.signals import request_finished, request_started
 from django.test.testcases import TransactionTestCase
 
-from unittest import mock
+import pytest
 
 from post_request_task.task import _discard_tasks, _stop_queuing_tasks
 
 from olympia.amo.celery import task
-from olympia.amo.tests import TestCase
 from olympia.amo.utils import utc_millesecs_from_epoch
+from olympia.amo.tests import wait_for_tasks
 
 
 fake_task_func = mock.Mock()
 
 
+def test_celery_routes_in_queues():
+    queues_in_queues = set([q.name for q in settings.CELERY_TASK_QUEUES])
+
+    # check the default queue is defined in CELERY_QUEUES
+    assert settings.CELERY_TASK_DEFAULT_QUEUE in queues_in_queues
+
+    # then remove it as it won't be in CELERY_ROUTES
+    queues_in_queues.remove(settings.CELERY_TASK_DEFAULT_QUEUE)
+
+    queues_in_routes = set(
+        [c['queue'] for c in settings.CELERY_TASK_ROUTES.values()])
+    assert queues_in_queues == queues_in_routes
+
+
+@task(ignore_result=False)
+def fake_task():
+    fake_task_func()
+    return 'foobar'
+
+
 @task
-def fake_task(**kw):
+def default_task():
     fake_task_func()
+    return 'foobar'
 
 
-class TestTaskTiming(TestCase):
+@task(track_started=True, ignore_result=False)
+def sleeping_task():
+    time.sleep(0.3)
 
-    def setUp(self):
-        patch = mock.patch('olympia.amo.celery.cache')
-        self.cache = patch.start()
-        self.addCleanup(patch.stop)
 
-        patch = mock.patch('olympia.amo.celery.statsd')
-        self.statsd = patch.start()
-        self.addCleanup(patch.stop)
+@pytest.mark.celery_worker_test
+def test_celery_worker_test_runs_through_worker():
+    result = sleeping_task.delay()
```

Comment:
I think this test case would be easier to understand if `sleeping_task` was receiving a `sleep_time` argument set to `0.3` here.